### DISCUSSION TITLE
logger = logging.getLogger(f"netbox.scripts.{script.full_name}")

### DISCUSSION THREAD

**Discussion by john2893** [netbox-community/netbox#15352] [Q&A] [answered]
Everything was working fine and all of a sudden get this:
I did enable logging and everything worked fine in the beginning and then this happened. I removed all the logging config and back to barebones and I still get this. 

I tried to remove the data input in the command line to see if I get any error but I get the same thing. 
Really appreciate any pointers please

This is my CLI:

Filename: ipnet.py
Class : NewIP

/opt/netbox/venv/bin/python3 /opt/netbox/netbox/manage.py runscript --loglevel debug ipnet.NewIP --commit --data '{"hostname": "testing.example.com", "address":"192.168.12.32/32" }

=======
```
Traceback (most recent call last):
  File "/opt/netbox/netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/netbox/venv/lib64/python3.9/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/opt/netbox/venv/lib64/python3.9/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/netbox/venv/lib64/python3.9/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/netbox/venv/lib64/python3.9/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/opt/netbox-3.6.4/netbox/extras/management/commands/runscript.py", line 95, in handle
    logger = logging.getLogger(f"netbox.scripts.{script.full_name}")
AttributeError: 'NoneType' object has no attribute 'full_name'


```

**Comment by candlerb** [comment]
Based on the backtrace, presumably you are using Netbox 3.6.4? Also, what does "all of a sudden" mean? Presumably something else changed - did you upgrade Netbox for example? Or make a change to your script?

However, without seeing the content of your script, it's impossible to see where the problem might be.

Making the smallest possible script that reproduces your problem is a very good way to approach this issue.  If you find that a minimal script works fine, but the script in question doesn't, then it's just a case of removing items from that script until you identify what's causing the problem.

However, if your minimal script still reproduces the problem, then post it here so we can reproduce the issue too.

I note that the Netbox [code](https://github.com/netbox-community/netbox/blob/v3.6.4/netbox/extras/management/commands/runscript.py#L77-L78) says:

```
        module_name, script_name = script.split('.', 1)
        module, script = get_module_and_script(module_name, script_name)
```

Therefore if script is None, it seems it was unable to load `ipnet.NewIP`. It could quite possibly be a syntax error in your script - although in that case Netbox could do with better error reporting. Again, having a script which reproduces the problem would allow the error reporting to be improved.

**Comment by candlerb** [comment, marked as answer]
I was able to reproduce your problem simply by putting a syntax error in a script: raised as #15353 so that this error handling can be improved in Netbox. I think the chances are that you've done the same.

You may get a more useful error message by running the following at the CLI:

```
/opt/netbox/venv/bin/python /opt/netbox/netbox/scripts/ipnet.py
```

If that doesn't show a syntax error, but instead shows a library loading error like `ModuleNotFoundError: No module named 'extras'`, then temporarily add the following four lines at the very top of your script, and try again:

```
import django, os, sys
sys.path.append('/opt/netbox/netbox')
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'netbox.settings')
django.setup()
```

Once you've found and fixed the problem, remove these lines.

**Reply by john2893** [reply]
It was an indentation error. 

My try and except block, the except line was indented 2 spaces forward. 
Thank you very very much! I spent so much time on this and I'm glad you responded. 

I used this method to figure the syntax error:

/opt/netbox/venv/bin/python /opt/netbox/netbox/scripts/ipnet.py
